Fixup installation of shared mpcdec library.
Based on gentoo patch.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---

diff -Nura musepack_src_r475.orig/libmpcdec/CMakeLists.txt musepack_src_r475/libmpcdec/CMakeLists.txt
--- musepack_src_r475.orig/libmpcdec/CMakeLists.txt	2010-08-21 12:55:01.000000000 -0300
+++ musepack_src_r475/libmpcdec/CMakeLists.txt	2011-08-17 15:53:06.126873408 -0300
@@ -1,7 +1,12 @@
 include_directories(${libmpc_SOURCE_DIR}/include)
 if(SHARED)
-  add_library(mpcdec SHARED huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  add_library(mpcdec_shared SHARED huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  set_target_properties(mpcdec_shared PROPERTIES OUTPUT_NAME mpcdec CLEAN_DIRECT_OUTPUT 1 VERSION 7.0.1 SOVERSION 7)
+  target_link_libraries(mpcdec_shared m)
+  install(TARGETS mpcdec_shared LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
 else(SHARED)
   add_library(mpcdec_static STATIC huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  set_target_properties(mpcdec_static PROPERTIES OUTPUT_NAME mpcdec CLEAN_DIRECT_OUTPUT 1)
+  target_link_libraries(mpcdec_static m)
+  install(TARGETS mpcdec_static LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
 endif(SHARED)
-
